CodeGeneration/Sources/generate-swift-syntax/InitSignature+Extensions.swift

```diff
@@ -15,10 +15,24 @@ import SwiftSyntaxBuilder
 import SyntaxSupport
 import Utils
 
-extension LayoutNode {
-  func generateInitializerDeclHeader(useDeprecatedChildName: Bool = false) -> SyntaxNodeString {
+extension InitSignature {
+  var compoundName: String {
+    let renamedArguments = children.map { child in
+        if child.isUnexpectedNodes {
+          return "_:"
+        } else {
+          return "\(child.labelDeclName):"
+        }
+      }.joined(separator: "")
+
+    return "init(leadingTrivia:\(renamedArguments)trailingTrivia:)"
+  }
+
+  func generateInitializerDeclHeader() -> SyntaxNodeString {
+    let stub = "public init"
```

Comment:
Do you think that extracting `public init` helps readability here? I would argue that it’s easier to understand if `public init` was inlined in the two locations it’s used.

Reply:
This is sort of preparing for a change in the next PR, where we sometimes end up using `init?` instead of `public init`. I can move this specific change into that commit, though, so we can figure out how we want to structure this once we get there.

Reply:
Oh, I see. Happy to leave it in that case.